File: backend/src/main/java/com/hotelmanagement/service/ReservationService.java
package com.hotelmanagement.service;

import java.util.List;

import com.hotelmanagement.dto.request.ReservationRequest;
import com.hotelmanagement.dto.response.ReservationResponse;
import com.hotelmanagement.enums.ReservationStatus;

public interface ReservationService {

    // Create
    ReservationResponse createReservation(ReservationRequest request);

    // Update
    ReservationResponse updateReservation(Integer reservationId,
                                          ReservationRequest request);

    // Delete
    void deleteReservation(Integer reservationId);

    // Get by ID
    ReservationResponse getReservationById(Integer reservationId);

    // Get all
    List<ReservationResponse> getAllReservations();

    // Search by Customer
    List<ReservationResponse> findByCustomer(Integer customerId);

    // Search by Room
    List<ReservationResponse> findByRoom(Integer roomId);

    // Search by Employee
    List<ReservationResponse> findByEmployee(Integer employeeId);

    // Search by Status
    List<ReservationResponse> findByStatus(ReservationStatus status);

}

File: frontend/src/services/reservationService.js
import api from "../api/api";

// =====================================================
// GET ALL RESERVATIONS
// =====================================================

export const getReservations = () => {
    return api.get("/reservations");
};


// =====================================================
// GET RESERVATION BY ID
// =====================================================

export const getReservationById = (id) => {
    return api.get(`/reservations/${id}`);
};


// =====================================================
// CREATE RESERVATION
// =====================================================

export const createReservation = (reservation) => {
    return api.post("/reservations", reservation);
};


// =====================================================
// UPDATE RESERVATION
// =====================================================

export const updateReservation = (id, reservation) => {
    return api.put(`/reservations/${id}`, reservation);
};


// =====================================================
// DELETE RESERVATION
// =====================================================

export const deleteReservation = (id) => {
    return api.delete(`/reservations/${id}`);
};


// =====================================================
// FIND RESERVATIONS BY CUSTOMER
// =====================================================

export const getReservationsByCustomer = (customerId) => {
    return api.get(`/reservations/customer/${customerId}`);
};


// =====================================================
// FIND RESERVATIONS BY ROOM
// =====================================================

export const getReservationsByRoom = (roomId) => {
    return api.get(`/reservations/room/${roomId}`);
};


// =====================================================
// FIND RESERVATIONS BY EMPLOYEE
// =====================================================

export const getReservationsByEmployee = (employeeId) => {
    return api.get(`/reservations/employee/${employeeId}`);
};


// =====================================================
// FIND RESERVATIONS BY STATUS
// =====================================================

export const getReservationsByStatus = (status) => {
    return api.get(`/reservations/status/${status}`);
};